### PR TITLE
refactor(npm): reorganize remapping built-in Node modules to remote URLs

### PR DESCRIPTION
**Pull request by bartlomieju**
Changes how built-in Node modules are mapped to polyfills
from "deno_std". Instead of intertwining this logic into Node
resolution logic, we map them to "NodeResolution::BuiltIn"
which are remapped to "deno_std" URLs in ProcState.

### CODE REVIEW

File: cli/node/mod.rs [hunk 1]
```diff
@@ -370,18 +373,9 @@ pub fn node_resolve(
...
-        )));
-      }
-
-      if let Some(resolved) = try_resolve_builtin_module(&specifier) {
-        return Ok(Some(ResolveResponse::Esm(resolved)));
-      } else {
-        return Err(generic_error(format!("Unknown module {}", specifier)));
+      if is_builtin_node_module(&specifier) {
+        let url = Url::parse(&format!("node:{}", specifier)).unwrap();
+        return Ok(Some(ResolveResponse::Esm(url)));
```
Comment (bartlomieju) [inline cli/node/mod.rs:378]: I feel we should have `ResolveResponse::Builtin` variant

Comment (dsherret) [reply cli/node/mod.rs:378]: I was thinking about refactoring this code earlier to use a different enum that's not deno_graph based, but then I forgot. IIRC, I don't believe this needs to use that `ResolveResponse` enum.

**Review by dsherret — APPROVED** [review]
LGTM

**Review by dsherret — APPROVED** [review]
Nice. LGTM

**Pull request merged**